### PR TITLE
feat: `decimal` type

### PR DESCRIPTION
**Pull request by nholik**
This adds a decimal 128 type along with tests for issue [94](https://github.com/plexinc/papr/issues/94)

### CODE REVIEW

File: benchmark/papr.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { papr, COLLECTIONS } from './setup.js';
 const sampleSchema = schema(
   {
     age: types.number(),
+    salary: types.number(),
```
Comment (joshuat) [inline benchmark/papr.js:8]:
```suggestion
    salary: types.decimal(),
```

File: src/types.ts [hunk 2]
```diff
@@ -360,6 +362,22 @@ export default {
    */
   constant,
 
+  /**
+   * Creates a IEEE 754  decimal-based 128 bit floating-point number type.
```
Comment (avaly) [inline src/types.ts:366]:
```suggestion
   * Creates a IEEE 754 decimal-based 128 bit floating-point number type.
```

File: src/types.ts [hunk 3]
```diff
@@ -360,6 +362,22 @@ export default {
...
+   * Useful for storing monetary values, scientific computations or any other number that requires high precision.
+   * @param [options] {GenericOptions}
```
Comment (avaly) [inline src/types.ts:368]:
```suggestion

   * @param [options] {GenericOptions}
```

File: src/__tests__/types.test.ts [hunk 4]
```diff
@@ -118,6 +118,35 @@ describe('types', () => {
       });
     });
 
+    describe('decimal', () => {
```
Comment (avaly) [inline src/__tests__/types.test.ts:121]: Please move this above, so that the sections are alphabetical: `date`, `decimal`, `enum`

**Review by avaly — CHANGES_REQUESTED** [review]
Thanks for the contribution. Just a few minor fixes before we can merge this.

File: benchmark/papr.js (hunk 1 above)
Comment (nholik) [reply benchmark/papr.js:8]: Decimal is used here so include an example monetary value in the benchmark and illustrate the change

**Comment by nholik** [comment]
> Thanks for the contribution. Just a few minor fixes before we can merge this.

Updated with the requested changes

File: benchmark/papr.js (hunk 1 above)
Comment (avaly) [reply benchmark/papr.js:8]: I think @joshuat's suggested change is still required here.

Comment (nholik) [reply benchmark/papr.js:8]: Not sure why this change should be done?  It was added specifically for this change with the new decimal type.  See the previous comment.  Let me know if I am misunderstanding.

Comment (avaly) [reply benchmark/papr.js:8]: In the Mongoose schema you defined `salary` as a `Decimal128` type. The Papr schema should match that type and use the new type this PR introduces.

Comment (nholik) [reply benchmark/papr.js]: oh my mistake, changed it back to number during testing and thought it was already decimal- was looking at it backwards.  Updated.

**Review by avaly — APPROVED** [review]

**Pull request merged**